hi-ml-histopathology/src/histopathology/utils/girder.py

```diff
@@ -0,0 +1,466 @@
+        items_jsons = result["item"]
+        if not items_jsons:
+            raise RuntimeError(f"No items found for query \"{text}\"")
+        elif len(items_jsons) > 1:
+            raise RuntimeError(f"More than one item found for query \"{text}\":\n{items_jsons}")
+        return Item(self, json=items_jsons[0])
+
+
+class Item:
+    def __init__(self, dsa: DigitalSlideArchive, id: Optional[str] = None, json: Optional[Dict] = None):
+        self._dsa = dsa
+
+        # Validate that only one of ID or the JSON dict have been passed
+        if id is not None and json is not None:
+            raise ValueError("Only the ID or the JSON object can be passed")
+        if id is None and json is None:
+            raise ValueError("An ID or JSON object must be passed")
+
+        if id is None:
+            id = json["_id"]  # type: ignore
+        self.id = id
+        self._json = json
+
+    @property
+    def url(self) -> str:
+        return urljoin(self._dsa.url, f"/#item/{self.id}")
+
+    def open(self) -> bool:
+        return webbrowser.open(self.url)
+
+    def add_annotation(self, annotation: Annotation) -> Dict:
+        response = self._dsa.post(
+            path="annotation",
+            parameters={"itemId": self.id},
+            json=annotation.as_json(),
+        )
+        return response
+
+
+class RunOutputs:
+    def __init__(
+            self,
+            run_id: str,
+            workspace_config_path: Optional[Path] = None,
+            force: bool = False,
+            ):  # noqa: E123
+        logging.info("Getting run \"%s\"...", run_id)
+        run = get_aml_run_from_run_id(run_id, workspace_config_path=workspace_config_path)
+        experiment = run.experiment
+        workspace = experiment.workspace
+
+        self.run = run
+        self.experiment = experiment
+        self.workspace = workspace
+        self.df = self.get_df(force)
+        self.tile_size = None
+
+    def get_df(self, force: bool) -> pd.DataFrame:
+        csv_name = f"test_output-{self.workspace.name}-{self.run.id}.csv"
+        local_csv_path = self.temp_dir / csv_name
+        if local_csv_path.is_file() and not force:
+            logging.info("Found cached CSV file")
+        else:
+            logging.info("Downloading outputs CSV...")
+            aml_exceptions = (
+                azureml.exceptions._azureml_exception.UserErrorException,
+                azureml._restclient.models.error_response.ErrorResponseException,
+            )
+            try:
+                self.run.download_file(AML_TEST_OUTPUTS_CSV, local_csv_path)
+            except aml_exceptions as e:
+                raise FileNotFoundError("Error downloading outputs file from run") from e
+        logging.info("Reading CSV file: %s...", local_csv_path)
+        return self._read_csv(local_csv_path)
+
+    @property
+    def temp_dir(self) -> Path:
+        return Path(tempfile.gettempdir())
+
+    def get_annotation_from_slide_data_frame(
```

Comment:
[Minor] Looks like this method could be static.

Reply:
Addressed in 58abef5.